libvips/iofuncs/type.c

```diff
@@ -563,10 +563,8 @@ vips_ref_string_new(const char *str)
 {
 	VipsArea *area;
 
-	if (!g_utf8_validate(str, -1, NULL))
-		str = "<invalid utf-8 string>";
-
-	area = vips_area_new((VipsCallbackFn) vips_area_free_cb, g_strdup(str));
+	area = vips_area_new((VipsCallbackFn) vips_area_free_cb,
+		g_utf8_make_valid(str, -1));
```

Comment:
I think this should be hoisted into a separate variable, as it computes the length on the original string a couple of lines below.
https://github.com/libvips/libvips/blob/75035c90713073aa74583694c99078e8a4ffc4aa/libvips/iofuncs/type.c#L571

(i.e. the length of the string should be computed on `g_utf8_make_valid(str, -1)` instead)